<a href="https://colab.research.google.com/github/BRIJNANDA1979/Segmenting-Kelp-Forests-/blob/main/Version_2_1_CNN_for_KELP.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Next Step is CNN training with Train 5635 1 band  and rejected 469 7Band data.
469*7 = 3293  rejected  Label 0
5635 Trained   Label 1

Total in training:  8928

In [ ]:
import pandas as pd
import os

In [ ]:
train_kelp_path = '/content/drive/MyDrive/train_kelp'

In [ ]:
train_kelp_files_list = os.listdir(train_kelp_path)

In [ ]:
test_satellite_path = '/content/drive/MyDrive/test_satellite'

In [ ]:
test_satellite_files_list = os.listdir(test_satellite_path)

In [ ]:
Predicted = os.listdir('/content/drive/MyDrive/Predicted_all_bands_files')

In [ ]:
len(Predicted)

6699

Rejected files  =   Test_satelleite - Predicted

469 7-Band files

In [ ]:
for file in Predicted:
  s = file.split('_')
  ss = s[0] + '_satellite.tif'
  #print(ss)
  if ss in test_satellite_files_list:
    test_satellite_files_list.remove(ss)

In [ ]:
len(test_satellite_files_list)

469

In [ ]:
rejected_files_list = test_satellite_files_list

In [ ]:
rejected_files_list[0:2]

['HP504630_satellite.tif', 'HR677407_satellite.tif']

In [ ]:
rejected_list_with_7_bands = []
for i in rejected_files_list:
  for j in range(1,8):
    rejected_list_with_7_bands.append(i.split('.')[0]  + str(j) + '.tif')


In [ ]:
rejected_list_with_7_bands[0:8]

['HP504630_satellite1.tif',
 'HP504630_satellite2.tif',
 'HP504630_satellite3.tif',
 'HP504630_satellite4.tif',
 'HP504630_satellite5.tif',
 'HP504630_satellite6.tif',
 'HP504630_satellite7.tif',
 'HR677407_satellite1.tif']

In [ ]:
len(train_kelp_files_list)

5635

In [ ]:
train_kelp_files_list[0:2]

['VN257517_kelp.tif', 'VN609253_kelp.tif']

In [ ]:
X_list = train_kelp_files_list + rejected_list_with_7_bands

In [ ]:
len(X_list)

8918

In [ ]:
df = pd.DataFrame(X_list, columns=['file_name'])

In [ ]:
df['file_name'][5633:5635]

5633    CW256294_kelp.tif
5634    CW399607_kelp.tif
Name: file_name, dtype: object

In [ ]:
df['file_name'][5635:5640]

5635    HP504630_satellite1.tif
5636    HP504630_satellite2.tif
5637    HP504630_satellite3.tif
5638    HP504630_satellite4.tif
5639    HP504630_satellite5.tif
Name: file_name, dtype: object

In [ ]:
label =[]
for i in range(0,5635):
  label.append(1)
for i in range(5635,8918):
  label.append(0)

In [ ]:
df['label'] = label

In [ ]:
df.head()

file_name  label
0  VN257517_kelp.tif      1
1  VN609253_kelp.tif      1
2  VN630519_kelp.tif      1
3  VN633188_kelp.tif      1
4  VN645015_kelp.tif      1

In [ ]:
df.tail()

file_name  label
8913  HL852722_satellite3.tif      0
8914  HL852722_satellite4.tif      0
8915  HL852722_satellite5.tif      0
8916  HL852722_satellite6.tif      0
8917  HL852722_satellite7.tif      0

In [ ]:
df['label'].value_counts()

1    5635
0    3283
Name: label, dtype: int64

In [ ]:
pip install rasterio

In [ ]:
s = 'GV801444_satellite2.tif'
s = s.split('_')


In [ ]:
int(s[1][-5])

2

In [ ]:
import rasterio

In [ ]:
XX = []
YY = []
for i in df['file_name']:
  #print(i)
  if i in train_kelp_files_list:
    with rasterio.open('/content/drive/MyDrive/train_kelp/' + i) as src:
      print(src.shape)
      print(src.meta)
      print(src.read(1))
      data_array = src.read(1)    ##### _kelp is 1 Band only
      XX.append(data_array)
      YY.append(1)
  else:
    s = i.split('_')
    ss = s[0] + '_satellite.tif'
    band = int(s[1][-5])
    with rasterio.open('/content/drive/MyDrive/test_satellite/' +ss) as src:
        data_array = src.read(band)     #### _satellite is 7 bands 1-7
        XX.append(data_array)
        YY.append(0)


/usr/local/lib/python3.10/dist-packages/rasterio/__init__.py:304: NotGeoreferencedWarning: Dataset has no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)


Streaming output truncated to the last 5000 lines.
(350, 350)
{'driver': 'GTiff', 'dtype': 'int8', 'nodata': None, 'width': 350, 'height': 350, 'count': 1, 'crs': None, 'transform': Affine(1.0, 0.0, 0.0,
       0.0, 1.0, 0.0)}
[[0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 ...
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]]
(350, 350)
{'driver': 'GTiff', 'dtype': 'int8', 'nodata': None, 'width': 350, 'height': 350, 'count': 1, 'crs': None, 'transform': Affine(1.0, 0.0, 0.0,
       0.0, 1.0, 0.0)}
[[0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 ...
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]]
(350, 350)
{'driver': 'GTiff', 'dtype': 'int8', 'nodata': None, 'width': 350, 'height': 350, 'count': 1, 'crs': None, 'transform': Affine(1.0, 0.0, 0.0,
       0.0, 1.0, 0.0)}
[[0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 ...
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]]
(350, 350)
{'driver': 'GTiff', 'dtype': 'int8', 'nodata': Non

In [ ]:
import numpy as np

In [ ]:
xx=  np.asarray(XX)
xx.shape

(8918, 350, 350)

In [ ]:
yy = np.asarray(YY)

In [ ]:
yy.shape

(8918,)

In [ ]:
xx = xx.reshape(xx.shape[0],xx.shape[1],xx.shape[2],1)

In [ ]:
xx.shape

(8918, 350, 350, 1)

In [ ]:
yy = yy.reshape(yy.shape[0],1)

In [ ]:
yy.shape

(8918, 1)

In [ ]:
X = xx
y= yy

In [ ]:
import keras

In [ ]:
model = keras.models.Sequential()

initializers = {

}

model.add(
    keras.layers.Conv2D(
        24, 5, input_shape=(350,350,1),
        activation='relu',
    )
)
model.add( keras.layers.MaxPooling2D(2) )
model.add(
    keras.layers.Conv2D(
        48, 5, activation='relu',
    )
)
model.add( keras.layers.MaxPooling2D(2) )
model.add(
    keras.layers.Conv2D(
        96, 5, activation='relu',
    )
)
model.add( keras.layers.Flatten() )
model.add( keras.layers.Dropout(0.2) )

model.add( keras.layers.Dense(
    1, activation='sigmoid',
    )
)

model.summary()

Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 conv2d (Conv2D)             (None, 346, 346, 24)      624       
                                                                 
 max_pooling2d (MaxPooling2D  (None, 173, 173, 24)     0         
 )                                                               
                                                                 
 conv2d_1 (Conv2D)           (None, 169, 169, 48)      28848     
                                                                 
 max_pooling2d_1 (MaxPooling  (None, 84, 84, 48)       0         
 2D)                                                             
                                                                 
 conv2d_2 (Conv2D)           (None, 80, 80, 96)        115296    
                                                                 
 flatten (Flatten)           (None, 614400)            0

In [ ]:
from sklearn.model_selection import train_test_split

In [ ]:
 X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.20, random_state=33)

In [ ]:
X_train.shape

(7134, 350, 350, 1)

In [ ]:
y_train.shape

(7134, 1)

In [ ]:
X_test.shape

(1784, 350, 350, 1)

In [ ]:
y_test.shape

(1784, 1)

In [ ]:
import tensorflow.keras.optimizers as optimizers

optimizer = optimizers.Adam(learning_rate=0.001)

In [ ]:
model.compile(loss='binary_crossentropy', optimizer=optimizer, metrics=['accuracy'])

In [ ]:
# Fit the model
model.fit(X, y, validation_split=0.33, epochs=1, batch_size=100)

60/60 [==============================] - 385s 6s/step - loss: 447.6172 - accuracy: 0.9655 - val_loss: 7087.2559 - val_accuracy: 0.7642


In [ ]:
model.fit(X_train, y_train, validation_data=(X_test,y_test), epochs=1, batch_size=100)

72/72 [==============================] - 449s 6s/step - loss: 36.3731 - accuracy: 0.9372 - val_loss: 0.4164 - val_accuracy: 0.9563


In [ ]:
prediction = model.predict(X_train)

223/223 [==============================] - 46s 205ms/step


In [ ]:
y_train[5]



array([1])

Using CNN let us verify already Predicted using CVI....


In [ ]:
Predicted_list_with_7_bands  = Predicted

In [ ]:
PX = []
XP = []    ### Predicted data
YP = []
for i in Predicted_list_with_7_bands:
  PX.append(i)
for i in PX:
  #print(i)
  s = i.split('_')
  ss = s[0] + '_satellite.tif'
  band = int(s[1][-5])
  with rasterio.open('/content/drive/MyDrive/test_satellite/' +ss) as src:
      data_array = src.read(band)     #### _satellite is 7 bands 1-7
      XP.append(data_array)
      YP.append(1)


/usr/local/lib/python3.10/dist-packages/rasterio/__init__.py:304: NotGeoreferencedWarning: Dataset has no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)


In [ ]:
XP = np.asarray(XP)

In [ ]:
XP.shape

(6699, 350, 350)

In [ ]:
XP = XP.reshape(XP.shape[0],XP.shape[1],XP.shape[2],1)

In [ ]:
XP.shape

(6699, 350, 350, 1)

In [ ]:
YP = np.asarray(YP)

In [ ]:
YP = YP.reshape(YP.shape[0],1)

In [ ]:
YP.shape

(6699, 1)

In [ ]:
Prediction_by_CNN = model.predict(XP)


210/210 [==============================] - 44s 207ms/step


In [ ]:
len(Prediction_by_CNN)

6699

In [ ]:
Prediction_by_CNN[0:10]

array([[0.       ],
       [0.       ],
       [0.       ],
       [0.       ],
       [0.9939565],
       [0.       ],
       [0.       ],
       [0.       ],
       [0.       ],
       [0.       ]], dtype=float32)

In [ ]:
count=0
for i in Prediction_by_CNN:
  if i[0] < 0.5 :
    count = count + 1
print(count)

5925


In [ ]:
Predicted_list_with_7_bands[0:10]               ##########3   GC205345_satellite1.tif

['VU293076_satellite2.tif',
 'VU293076_satellite3.tif',
 'VU293076_satellite4.tif',
 'VU293076_satellite5.tif',
 'VU293076_satellite6.tif',
 'VU293076_satellite7.tif',
 'WI130359_satellite1.tif',
 'WI130359_satellite2.tif',
 'WI130359_satellite3.tif',
 'WI130359_satellite4.tif']

In [ ]:
loc =0
for file in Predicted_list_with_7_bands:
  if(file == 'GC205345_satellite1.tif'):
    print(file)
    print(loc)
    print(Prediction_by_CNN[loc])
    break

  loc = loc + 1

GC205345_satellite1.tif
573
[0.]


In [ ]:
loc =0                                            ###########3 AC799254_satellite.tif
for file in Predicted_list_with_7_bands:
  if(file == 'AC799254_satellite.tif'):
    print(file)
    print(loc)
    break
  loc = loc + 1

In [ ]:
Df_Predictions = pd.DataFrame()
Df_Predictions['images'] = Predicted_list_with_7_bands
label_list = []
for file in Predicted_list_with_7_bands:
  label_list.append(1)
Df_Predictions['Y_Predicted_by_CVI'] = label_list
label_CNN =[]
for i in Prediction_by_CNN:
  if i[0] < 0.5 :
    label_CNN.append(0)
  else:
    label_CNN.append(1)
Df_Predictions['Y_Predicted_by_CNN'] = label_CNN
Df_Predictions.to_csv('/content/drive/MyDrive/Kelp_Predicted_by_CNN/Predictions_by_CNNversion_2_.csv')

In [ ]:
Df_Predictions.describe()

Y_Predicted_by_CVI  Y_Predicted_by_CNN
count              6699.0         6699.000000
mean                  1.0            0.115540
std                   0.0            0.319696
min                   1.0            0.000000
25%                   1.0            0.000000
50%                   1.0            0.000000
75%                   1.0            0.000000
max                   1.0            1.000000

In [ ]:
Df_Predictions.head()

images  Y_Predicted_by_CVI  Y_Predicted_by_CNN
0  VU293076_satellite2.tif                   1                   0
1  VU293076_satellite3.tif                   1                   0
2  VU293076_satellite4.tif                   1                   0
3  VU293076_satellite5.tif                   1                   0
4  VU293076_satellite6.tif                   1                   1

After running CNN on predictions 6699 made by CVI,
5925 ----> 0
774 ------>  1

So need to remove 5925 from 6699 predicetd

In [ ]:
Df_Predictions['Y_Predicted_by_CNN'].value_counts()

0    5925
1     774
Name: Y_Predicted_by_CNN, dtype: int64

In [ ]:
Refined_list_after_CNN = []
for row in Df_Predictions.itertuples():
  if row.Y_Predicted_by_CNN == 1:
    Refined_list_after_CNN.append(row.images)

In [ ]:
len(Refined_list_after_CNN)

774

In [ ]:
Refined_list_after_CNN[0:2]

['VU293076_satellite6.tif', 'WI130359_satellite6.tif']

In [83]:
import shutil

In [86]:
for i in Refined_list_after_CNN:
  s = i.split('_')
  ss = s[0] + '_satellite.tif'
  band = int(s[1][-5])
  with rasterio.open('/content/drive/MyDrive/test_satellite/' +ss) as src:
        data_array = src.read(band)     #### _satellite is 7 bands 1-7
       # Register GDAL format drivers and configuration options with a
        # context manager.
        with rasterio.Env():

          # Write an array as a raster band to a new 8-bit file. For
          # the new file's profile, we start with the profile of the source
          profile = src.profile

          # And then change the band count to 1, set the
          # dtype to uint8, and specify LZW compression.
          profile.update(
                        dtype=rasterio.uint8,
                        count=1,
                        compress='lzw')
          with rasterio.open('/content/drive/MyDrive/KELP_Submission'+ '/'+ i, 'w', **profile) as dst:
              dst.write(data_array.astype(rasterio.uint8), 1)
          # At the end of the ``with rasterio.Env()`` block, context
          # manager exits and all drivers are de-registered.



/usr/local/lib/python3.10/dist-packages/rasterio/__init__.py:304: NotGeoreferencedWarning: Dataset has no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
/usr/local/lib/python3.10/dist-packages/rasterio/__init__.py:314: NotGeoreferencedWarning: The given matrix is equal to Affine.identity or its flipped counterpart. GDAL may ignore this matrix and save no geotransform without raising an error. This behavior is somewhat driver-specific.
  dataset = writer(


TI176604_satellite6.tif   unexpected file

NEW CNN

use KELP as Train Ground Truth 5635 images
use Rejected 5925 images from last CNN prediction

hence 5635: 1
      5925: 0

      Total: 11560

In [9]:
import os
import pandas as pd

In [6]:
XX_train_list= os.listdir('/content/drive/MyDrive/train_kelp')

In [7]:
len(  XX_train_list)

5635

In [8]:
XX_train_list[0:2]

['VN257517_kelp.tif', 'VN609253_kelp.tif']

In [10]:
Df_Predictions= pd.read_csv('/content/drive/MyDrive/Kelp_Predicted_by_CNN/Predictions_by_CNNversion_2_.csv')

In [11]:
DF = pd.DataFrame()
DF  = Df_Predictions
DF.head()

Unnamed: 0                   images  Y_Predicted_by_CVI  Y_Predicted_by_CNN
0           0  VU293076_satellite2.tif                   1                   0
1           1  VU293076_satellite3.tif                   1                   0
2           2  VU293076_satellite4.tif                   1                   0
3           3  VU293076_satellite5.tif                   1                   0
4           4  VU293076_satellite6.tif                   1                   1

In [12]:
DF.describe()

Unnamed: 0  Y_Predicted_by_CVI  Y_Predicted_by_CNN
count  6699.000000              6699.0         6699.000000
mean   3349.000000                 1.0            0.115540
std    1933.979059                 0.0            0.319696
min       0.000000                 1.0            0.000000
25%    1674.500000                 1.0            0.000000
50%    3349.000000                 1.0            0.000000
75%    5023.500000                 1.0            0.000000
max    6698.000000                 1.0            1.000000

In [13]:
rejected_by_Last_CNN = []
for row in DF.itertuples():
  if row.Y_Predicted_by_CNN == 0:
    rejected_by_Last_CNN.append(row.images)

In [14]:
len(rejected_by_Last_CNN)

5925

In [15]:
rejected_by_Last_CNN[0:2]

['VU293076_satellite2.tif', 'VU293076_satellite3.tif']

In [16]:
All_for_training_new_CNN = XX_train_list + rejected_by_Last_CNN

In [17]:
len(All_for_training_new_CNN)

11560

In [19]:
pip install rasterio

In [20]:
import rasterio

In [21]:
XXX =[]
YYY =[]
for i in XX_train_list:    #### for Train Kelp data
    with rasterio.open('/content/drive/MyDrive/train_kelp/' +i) as src:
      data_array = src.read(1)
      XXX.append(data_array)
      YYY.append(1)

/usr/local/lib/python3.10/dist-packages/rasterio/__init__.py:304: NotGeoreferencedWarning: Dataset has no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)


In [22]:
for i in rejected_by_Last_CNN:
  s = i.split('_')
  ss = s[0] + '_satellite.tif'
  band = int(s[1][-5])
  #### for Rejected sattelite 7 band data
  with rasterio.open('/content/drive/MyDrive/test_satellite/' +ss) as src:
    data_array = src.read(band)
    XXX.append(data_array)
    YYY.append(0)

In [23]:
import numpy as np

In [24]:
XXX = np.array(XXX)
YYY = np.array(YYY)

In [25]:
XXX.shape

(11560, 350, 350)

In [26]:
XXX = np.expand_dims(XXX, axis=3)

In [27]:
XXX.shape

(11560, 350, 350, 1)

In [28]:
YYY= np.expand_dims(YYY, axis=1)

In [29]:
YYY.shape

(11560, 1)

In [30]:
X=XXX
y=YYY

In [33]:
import keras
from keras.models import Sequential
from keras.layers import Dense, Dropout, Flatten


In [34]:
model = keras.models.Sequential()

initializers = {

}

model.add(
    keras.layers.Conv2D(
        24, 5, input_shape=(350,350,1),
        activation='relu',
    )
)
model.add( keras.layers.MaxPooling2D(2) )
model.add(
    keras.layers.Conv2D(
        48, 5, activation='relu',
    )
)
model.add( keras.layers.MaxPooling2D(2) )
model.add(
    keras.layers.Conv2D(
        96, 5, activation='relu',
    )
)
model.add( keras.layers.Flatten() )
model.add( keras.layers.Dropout(0.2) )

model.add( keras.layers.Dense(
    1, activation='sigmoid',
    )
)

model.summary()

Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 conv2d (Conv2D)             (None, 346, 346, 24)      624       
                                                                 
 max_pooling2d (MaxPooling2D  (None, 173, 173, 24)     0         
 )                                                               
                                                                 
 conv2d_1 (Conv2D)           (None, 169, 169, 48)      28848     
                                                                 
 max_pooling2d_1 (MaxPooling  (None, 84, 84, 48)       0         
 2D)                                                             
                                                                 
 conv2d_2 (Conv2D)           (None, 80, 80, 96)        115296    
                                                                 
 flatten (Flatten)           (None, 614400)            0

In [35]:
from sklearn.model_selection import train_test_split

In [36]:
 X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.20, random_state=33)

In [37]:
X_train.shape

(9248, 350, 350, 1)

In [38]:
X_test.shape

(2312, 350, 350, 1)

In [39]:
y_train.shape

(9248, 1)

In [40]:
y_test.shape

(2312, 1)

In [41]:
import tensorflow.keras.optimizers as optimizers

optimizer = optimizers.Adam(learning_rate=0.001)

In [42]:
model.compile(loss='binary_crossentropy', optimizer=optimizer, metrics=['accuracy'])

In [43]:
# Fit the model
model.fit(X, y, validation_split=0.33, epochs=1, batch_size=100)

78/78 [==============================] - 539s 7s/step - loss: 17.3493 - accuracy: 0.9424 - val_loss: 0.7491 - val_accuracy: 0.9874


In [45]:
predicted_as_1_by_Last_CNN = []
for row in DF.itertuples():
  if row.Y_Predicted_by_CNN == 1:
    predicted_as_1_by_Last_CNN.append(row.images)


In [48]:
predicted_as_1_by_Last_CNN[0:2]

['VU293076_satellite6.tif', 'WI130359_satellite6.tif']

In [49]:
len(predicted_as_1_by_Last_CNN)

774

In [50]:
PXXX = []
PYYY = []
for i in predicted_as_1_by_Last_CNN:
  s = i.split('_')
  ss = s[0] + '_satellite.tif'
  band = int(s[1][-5])
  #### for Rejected sattelite 7 band data
  with rasterio.open('/content/drive/MyDrive/test_satellite/' +ss) as src:
    data_array = src.read(band)
    PXXX.append(data_array)
    PYYY.append(1)


/usr/local/lib/python3.10/dist-packages/rasterio/__init__.py:304: NotGeoreferencedWarning: Dataset has no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)


In [62]:
PX = PXXX
PY = PYYY

In [63]:
PX = np.array(PX)
PX.shape

(774, 350, 350)

In [64]:
prediction_from_new_CNN = model.predict(PX)

25/25 [==============================] - 6s 219ms/step


In [68]:
len(prediction_from_new_CNN)

774

In [69]:
prediction_from_new_CNN[0:10]

array([[1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.]], dtype=float32)

In [72]:
count=0
for i in prediction_from_new_CNN:
  if i[0] < 0.5 :
    count = count + 1
print(count)

3


In [73]:
label=[]
for i in prediction_from_new_CNN:
  if(i[0] < 0.5):
    label.append(0)
  else:
    label.append(1)


In [76]:
predicted_as_1_by_Last_CNN[0:2]

['VU293076_satellite6.tif', 'WI130359_satellite6.tif']

In [75]:
len(predicted_as_1_by_Last_CNN)

774

In [77]:
df_new_CNN = pd.DataFrame(predicted_as_1_by_Last_CNN, columns=['images'])

In [78]:
df_new_CNN['label'] = label

In [79]:
df_new_CNN.head()

images  label
0  VU293076_satellite6.tif      1
1  WI130359_satellite6.tif      1
2  NV982544_satellite6.tif      1
3  FP277965_satellite6.tif      1
4  WB585968_satellite6.tif      1

In [80]:
df_new_CNN.describe()

label
count  774.000000
mean     0.996124
std      0.062177
min      0.000000
25%      1.000000
50%      1.000000
75%      1.000000
max      1.000000

In [82]:
df_new_CNN['label'].value_counts()

1    771
0      3
Name: label, dtype: int64

In [83]:
df_new_CNN[df_new_CNN['label']==0]

images  label
27   ZY575159_satellite6.tif      0
105  NR161645_satellite6.tif      0
352  KU431495_satellite6.tif      0

In [85]:
if 'TI176604_satellite6.tif' in predicted_as_1_by_Last_CNN:
  print('yes')


yes
